cluster/config_test.go

```diff
@@ -28,6 +28,18 @@ func TestBuildContainerConfig(t *testing.T) {
 	config = BuildContainerConfig(container.Config{Env: []string{"test=true", "constraint:test==true", "affinity:container==test"}}, container.HostConfig{}, network.NetworkingConfig{})
 	assert.Len(t, config.Env, 1)
 	assert.Len(t, config.Labels, 2)
+
+	networkingConfig := network.NetworkingConfig{
+		EndpointsConfig: map[string]*network.EndpointSettings{
+			"testnet1":           &network.EndpointSettings{},
+			"nodename/testnet2":  &network.EndpointSettings{},
+			"nodename/test/net3": &network.EndpointSettings{},
+		},
+	}
+	config = BuildContainerConfig(container.Config{}, container.HostConfig{}, networkingConfig)
+	assert.Contains(t, config.NetworkingConfig.EndpointsConfig, "testnet1")
+	assert.Contains(t, config.NetworkingConfig.EndpointsConfig, "testnet2")
+	assert.Contains(t, config.NetworkingConfig.EndpointsConfig, "test/net3")
```

Comment:
does `assert.Contains` not mean that the string `"nodename/testnet2"` would pass the assertion?

Reply:
`config.NetworkingConfig.EndpointsConfig` is a map, so we're checking for whether or not this exact key is in the map.